### PR TITLE
Cache start

### PR DESCRIPTION
**Pull request by wuminzhe**
1. optional --data-dir <dir>, default is ~/.bridger
2. cache start
    if cache not exist, take ethereum latest block number as start
3. set-start --block 10000
4. take effect immediately after set-start

#32 

5. restart refactor

### CODE REVIEW

**Comment by HackFisher** [comment]
@wuminzhe Can we automatically update the start according to the latest redeemed tx? Seems we can only set the start manually in this PR?

**Comment by wuminzhe** [comment]
@HackFisher it take ethereum lastest block as start if not cache not exist. and save the last scanned block to cache after a scan.

**Comment by HackFisher** [comment]
> @HackFisher it take ethereum lastest block as start if cache not exist. and save the last scanned block to cache after a scan.

I think we should use last redeemed tx block as start, not the latest scanned block number.

Using latest scanned block number introduce possibilities that we might miss some ethereum tx which need redeem.

**Comment by wuminzhe** [comment]
yes, I will change it to the last redeemed tx block

**Comment by wuminzhe** [comment]
@HackFisher  last redeemed tx from chain or local?

File: src/service/redeem.rs [hunk 1]
```diff
@@ -151,6 +171,39 @@ impl RedeemService {
...
         info!("      Redeemed ethereum tx {:?} with extrinsic {:?}", tx.enclosed_hash(), hash);
 
+        // 3. Update cache
+        RedeemService::set_last_redeemed(data_dir, tx.block)?;
+        Ok(())
+    }
+
+    const LAST_REDEEMED_CACHE_FILE_NAME: &'static str = "last-redeemed";
+
+    /// get_ethereum_start
```
Comment (HackFisher) [inline src/service/redeem.rs:181]: update comment

File: src/service/redeem.rs [hunk 2]
```diff
@@ -151,6 +171,39 @@ impl RedeemService {
...
+        let mut file = File::open(filepath)?;
+        let mut buffer = String::new();
+        file.read_to_string(&mut buffer)?;
+        match buffer.trim().parse() {
+            Ok(start) => Ok(start),
+            Err(e) => Err(Bridger(e.to_string()))
+        }
+    }
+
+    /// set_ethereum_start
```
Comment (HackFisher) [inline src/service/redeem.rs:201]: update comment

**Review by HackFisher — CHANGES_REQUESTED** [review]
@wuminzhe Please update the sample config files in .maintain too, thanks

**Review by HackFisher — APPROVED** [review]

**Pull request merged**